pkg/trace/agent/agent.go

```diff
@@ -55,6 +55,9 @@ type Agent struct {
 	obfuscator     *obfuscate.Obfuscator
 	cardObfuscator *ccObfuscator
 
+	// FilterSpan will be called on all spans, if non-nil. If it returns true, the span will be deleted before processing.
+	FilterSpan func(*pb.Span) bool
```

Comment:
I think `DiscardSpan` is a more explanatory name.

Reply:
It looks like this can actually be removed? I don't see a reference to the `FilterSpan` type anywhere.